software/contrib/update_firmware.py

```diff
@@ -0,0 +1,34 @@
+from machine import bootloader, reset
+from europi_script import EuroPiScript
+from europi import b1, b2, oled
+from os import remove
+
+
+class UpdateFirmware(EuroPiScript):
+    @classmethod
+    def display_name(cls):
+        """Push this script to the end of the menu."""
+        return "~Update Firmware"
+        
+    def back(self):
+        try:
+            remove('saved_state_BootloaderMenu.txt')	#This file needs to be removed so that the menu is returned to rather than this script once it's entered bootloader mode
+        except OSError:
+            print('OSError: File Not Found')
+        reset()	#Restart the module so that it will boot back into the menu
+    
+    def enter_bootloader(self):
+        bootloader()	#Enter bootloader mode
+        
+    def main(self):
+        b1.handler(self.back)	#Button 1 returns to the menu system
+        b2.handler(self.enter_bootloader)	#Button 2 sends the module into bootloader mode
+        
+        oled.fill(0)
+        oled.text('B1 = Back',0,6)
+        oled.text('B2 = Bootloader',0,18)
```

Comment:
[optional] This seems like a good place to use `Display.centre_text()`.

Reply:
I think that when the start of the text is two options of identical length (B1 and B2) it looks better when it's left aligned. Centred text for something like this would read more as one single statement rather than two options to me, but it is subjective!

Reply:
Yeah, you are right. I was thinking of the asthetics of the code itself, using centre_text would have removed the coordinates. I like your way better.